RInAppMessaging/Classes/Utilities/AtomicWrapper.swift

```diff
@@ -15,12 +17,12 @@ struct AtomicGetSet<Value> {
             return queue.sync { value }
         }
         set {
-            queue.sync { value = newValue }
+            queue.sync(flags: .barrier) { value = newValue }
```

Comment:
[obj.io](https://www.objc.io/issues/2-concurrency/low-level-concurrency-apis/) suggests that barrier calls should be async:
>When you use concurrent queues, make sure that all barrier calls are async . If you're using dispatch_barrier_sync you'll quite likely get yourself (or rather: your code) into a deadlock. Writes need a barrier, and can be async.

But there are many other blogs/tutorials about GCD that use `sync` with barrier so I'm not sure.

Is there a reason that sync cannot cause us a problem?
What queue calls the AtomicGetSet?

Reply:
Each AtomicGetSet property gets its own queue but setter can be called by any thread. There's no way for a deadlock here. I used sync to be sure that calling getter immediately after set will return expected value.